### PR TITLE
net-dns/knot: add 3.4.3, drop 3.4.2

### PR DESCRIPTION
**Pull request by PPN-SD**
https://www.knot-dns.cz/2024-12-06-version-343.html

---

Please check all the boxes that apply:

- [X] I can submit this contribution in agreement with the [Copyright Policy](https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin).
- [X] I have certified the above via adding a `Signed-off-by` line to *every* commit in the pull request.
- [X] This contribution has not been created with the assistance of Natural Language Processing artificial intelligence tools, in accordance with the [AI policy](https://wiki.gentoo.org/wiki/Project:Council/AI_policy).
- [X] I have run `pkgcheck scan --commits --net` to check for issues with my commits.

Please note that all boxes must be checked for the pull request to be merged.

### CODE REVIEW

File: net-dns/knot/knot-3.4.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	test? (
+		pkcs11? ( dev-libs/softhsm )
+	)
+"
+
+src_prepare() {
+	default
+
+	# these tests call this daemon file knot/server/dthreads.h
+	if use test && use !daemon; then
```
Comment (thesamesam) [inline net-dns/knot/knot-3.4.3.ebuild]: Conditional sed is already a bit of a shame but conditional eautoreconf is a bit odd (especially as to be completely correct, we'd really need to then set AUTOTOOLS_AUTO_DEPEND=no and so on).

Can we somehow skip them via a patch or something instead?

Comment (thesamesam) [reply net-dns/knot/knot-3.4.3.ebuild]: Yes, please do for now (include a reference to that bug in there).

Comment (PPN-SD) [reply net-dns/knot/knot-3.4.3.ebuild]: Not so easy with a patch without automake.

But, this does the job ... :
```
src_test() {
    sed \
        -e 's:contrib/test_atomic$(EXEEXT)::' \
        -e 's:contrib/test_spinlock$(EXEEXT)::' \
        -i tests/Makefile || die
    rm tests/contrib/test_{atomic,spinlock}.c || die

    default
}
```

Comment (thesamesam) [reply net-dns/knot/knot-3.4.3.ebuild]: Why not just unconditionally eautoreconf, or replace them with stub C files instead which just exit 77?

Comment (PPN-SD) [reply net-dns/knot/knot-3.4.3.ebuild]: I misunderstanded about eautoreconf. Here's patch with skipped tests.

File: net-dns/knot/files/knot-3.4.3-skip_atomic_spinlock_tests.patch [hunk 2]
```diff
@@ -0,0 +1,257 @@
+https://gitlab.nic.cz/knot/knot-dns/-/issues/946
```
[the remark this answers is not in the document]
Comment (PPN-SD) [reply net-dns/knot/files/knot-3.4.3-skip_atomic_spinlock_tests.patch]: I think it's ok. Thanks for your time with a beginner :+1:

[the remark this answers is not in the document]
Comment (thesamesam) [reply net-dns/knot/files/knot-3.4.3-skip_atomic_spinlock_tests.patch]: You're very welcome! Thank you for the good-quality contributions :)

**Pull request closed without merging**